db/collection.go

```diff
@@ -101,9 +94,11 @@ func (c *collection) newFetcher() fetcher.Fetcher {
 func (db *db) createCollection(
 	ctx context.Context,
 	txn datastore.Txn,
-	desc client.CollectionDescription,
-	schema client.SchemaDescription,
+	def client.CollectionDefinition,
 ) (client.Collection, error) {
+	schema := def.Schema
+	desc := def.Description
```

Comment:
question: Why reallocate when you've already passed a copy of the definition? Using `def` directly would be clearer and you could assign to `col.def` directly from it.

Reply:
It is a struct, and its children are structs, mutating them does not mutate the one on the parent, the parent must instead later be overwritten with a new inner.

```
// foo.bar.foobar is false
foo.bar.foobar = true
// foo.bar.foobar is still false, a new (and unreachable) bar instance has value true
```

Reply:
I don't think that's true: https://go.dev/play/p/FHce-ajgFyr

Reply:
:laughing: Maybe not, but something was very broken before I made that change. It is very deliberate and it doesnt work without it.  Could be something similar in the more complex real-world code.

Reply:
I just tested it and all tests are passing. Not sure what would have been broken.

Reply:
I cannot see what you have changed, most likely it was schema that was important here, as that is what is being mutated most.